### DISCUSSION TITLE
FEM thermal radiation

### DISCUSSION THREAD

**Discussion by Gallasd** [idaholab/moose#21595] [Q&A General] [answered]
Hi, I want to use moose as my solver for structural optimization. The optimization problem is first to determine the governing equations.
I study heat conduction under radiation boundaries. 
The governing equation is the heat conduction equation, and the boundary condition is the radiation boundary, that is, the fourth power of temperature, similar to this:
![image](https://user-images.githubusercontent.com/107783970/179238866-ecd1afd1-51bc-430e-8ff1-09e8c7640ac1.png)
If it is a convection boundary, the weighted margin method is directly used, and the integration is sufficient. For the radiation boundary, this highly nonlinear problem is very difficult to deal with by the finite element method.
Many literatures simplify the radiation boundary to a certain extent, such as linear approximation of radiation heat transfer and equivalent heat transfer. But there must be a general simplification, because software cannot simplify equations for every simplified physical problem.
So I would like to ask you how moose performs finite element processing of radiation boundaries.

**Comment by GiudGiud** [comment, marked as answer]
Hello

MOOSE has a non-linear solver so we typically can solve these boundary conditions exactly. We have various objects for dealing with radiative heat transfer:
https://mooseframework.inl.gov/source/actions/ThermalContactAction.html
is for radiation / conduction and convection at interfaces, which may be disjoint. 

https://mooseframework.inl.gov/syntax/GrayDiffuseRadiation/index.html
is for "exact" treatment of radiation using ray tracing. This is more expensive to set up. 

https://mooseframework.inl.gov/source/bcs/ADFunctionRadiativeBC.html
https://mooseframework.inl.gov/source/bcs/ADInfiniteCylinderRadiativeBC.html
are meant to have radiation boundary conditions at the outer boundary of the mesh (or of the temperature variable in the mesh)
They effectively implement different approximation (radiative surface shape of a plane in front or a cylinder in front respectively)

Guillaume

**Reply by Gallasd** [reply]
Hello, maybe you didn't understand my question, I hope you can read it patiently.
Q1:I'm not sure how you deal with the radiation boundary conditions above on finite elements, such as ADFunctionRadiation. Here is a document I found that puts the effect of radiation boundaries into the stiffness matrix, simplifying this highly nonlinear question. 
![c01dd8598f91fa4410c9caebb6918a8](https://user-images.githubusercontent.com/107783970/179336268-6c06d02d-ddb3-4bcf-9859-346170fca9e8.png)
Theoretically, the role of the radiation boundary should be in the load matrix.
So I would like to confirm with you how moose calculates the ADFunctionRadiation, which is a problem of the fourth power of heat flux, and whether it adopts the method used in the above literature.

Q2: I have another question. In the calculation example provided by ADFunctionRadiation, there is such a passage:
```
# If we assume that epsilon*sigma*(T_inf^4-T_s^4) is approximately equal to
# epsilon*sigma*4*T_inf^3*(T_inf-T_s), that form is equivalent to
# h*(T_inf-T_s), the convective flux bc. So, the radiative and convective
# flux bcs should give nearly the same answer if the leading terms are equal.
```
I don't understand why this can be so simplified, why T_inf can be brought up, and why there is an extra 4. In addition, is the radiation problem so simplified in moose?

**Reply by GiudGiud** [reply]
Q1: no we dont do this approach. We form the residual with the current solution values for all variables, then either do Newton or PJFNK. Forming this load and stiffness matrix is for a custom numerical scheme

Q2: Factorize (T_inf^4-T_s^4) by (T_inf-T_s) and look at the resulting polynomial. You can approximate T_s by T_inf in the rest of the factorization and get what the text says

The radiation problem is not so simplified in MOOSE, this is just a sanity check. 
You may look at the code at your convenience to convince yourself

**Comment by Gallasd** [comment]
Hi, thanks for your reply.
I learned about the radiation part of ANSYS and moose, and basically understood the principle of radiation calculation. It is found that the radiation problem for a surface, that is, the temperature without surface radiation exchange, should also be solved by the theory of Gray Diffuse Radiative Exchange.
This is my geometry, obviously, there is no heat radiation between the surfaces:
![image](https://user-images.githubusercontent.com/107783970/179401302-21d5bd7e-c27a-481e-932c-12ec72f919e8.png)

After carefully studying the radiation theory in moose, I think that in moose,The radiation boundary condition becomes a constant heat flow boundary condition,here J is a constant in this equation：
![image](https://user-images.githubusercontent.com/107783970/179401329-c9f025e2-f386-4840-ad21-b2ea1fdcd6e4.png)

For J  in the formula, calculate by the following formula:
![image](https://user-images.githubusercontent.com/107783970/179401575-61ad68fa-a0ce-4ff8-a650-45d4130650c4.png)

Here, through the Newton or PJFNK you mentioned, to solve the nonlinear equation system of  Equation 2.Constantly exchange information with the thermal conductivity control equation, perform iterations, and finally determine the converged J and T.

So Newton or PJFNK solve for the variables on the radiation boundary, not the overall variables？
thank you for your reply！

**Reply by GiudGiud** [reply]
It's not a local solve on the boundary, it's for all variables at the same time. The boundary temperature is converged at the same time, there is no separate treatment afaik
The equations for each implementation of the radiation BC are in their respective documentation. It's indeed a heat flux, but it's not constant. Most of them vary as T^4 - T_inf^4
